core/include/jiminy/core/stepper/LieGroup.h

```diff
@@ -0,0 +1,82 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+///
+/// \brief      Handling of the joint configuration q as part of a Lie Group, not a vector space.
+///
+/// \details    The state of the system, (q, v), lies in a Lie Group, while its derivative
+///             (v, a) is a tangent vector to this group: the classes in this file are
+///             meant to handle this.
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef JIMINY_LIE_GROUP_H
+#define JIMINY_LIE_GROUP_H
+
+#include "jiminy/core/Types.h"
+
+namespace jiminy
+{
+    class Robot;
+
+    class State
```

Comment:
You need to forward-declare `StateDerivative` before defining `State` class since you are declaring `+` operator, which requires knowing this type exists.